### PR TITLE
Fix handling of one_manager_agent_env pytest marker in System tests

### LINKED ISSUE #4763: Wazuh `one_manager_agent` `system tests` are failing due to KeyError: 'managers' while is executed using one_manager_agent_env mark
Running [system test in v4.8.0-alpha1](https://github.com/wazuh/wazuh/issues/20671)
In `one_manager_agent` environment test could not be executed using the mark one_manager_agent_env showing the following error:

<details>

```
test_cluster/test_correct_merged_file_generation/test_correct_merged_file_generation.py 
INTERNALERROR> Traceback (most recent call last):
INTERNALERROR>   File "/home/akim/.local/lib/python3.10/site-packages/_pytest/main.py", line 268, in wrap_session
INTERNALERROR>     session.exitstatus = doit(config, session) or 0
INTERNALERROR>   File "/home/akim/.local/lib/python3.10/site-packages/_pytest/main.py", line 322, in _main
INTERNALERROR>     config.hook.pytest_runtestloop(session=session)
INTERNALERROR>   File "/usr/lib/python3/dist-packages/pluggy/hooks.py", line 286, in __call__
INTERNALERROR>     return self._hookexec(self, self.get_hookimpls(), kwargs)
INTERNALERROR>   File "/usr/lib/python3/dist-packages/pluggy/manager.py", line 92, in _hookexec
INTERNALERROR>     return self._inner_hookexec(hook, methods, kwargs)
INTERNALERROR>   File "/usr/lib/python3/dist-packages/pluggy/manager.py", line 83, in <lambda>
INTERNALERROR>     self._inner_hookexec = lambda hook, methods, kwargs: hook.multicall(
INTERNALERROR>   File "/usr/lib/python3/dist-packages/pluggy/callers.py", line 208, in _multicall
INTERNALERROR>     return outcome.get_result()
INTERNALERROR>   File "/usr/lib/python3/dist-packages/pluggy/callers.py", line 80, in get_result
INTERNALERROR>     raise ex[1].with_traceback(ex[2])
INTERNALERROR>   File "/usr/lib/python3/dist-packages/pluggy/callers.py", line 187, in _multicall
INTERNALERROR>     res = hook_impl.function(*args)
INTERNALERROR>   File "/home/akim/.local/lib/python3.10/site-packages/_pytest/main.py", line 347, in pytest_runtestloop
INTERNALERROR>     item.config.hook.pytest_runtest_protocol(item=item, nextitem=nextitem)
INTERNALERROR>   File "/usr/lib/python3/dist-packages/pluggy/hooks.py", line 286, in __call__
INTERNALERROR>     return self._hookexec(self, self.get_hookimpls(), kwargs)
INTERNALERROR>   File "/usr/lib/python3/dist-packages/pluggy/manager.py", line 92, in _hookexec
INTERNALERROR>     return self._inner_hookexec(hook, methods, kwargs)
INTERNALERROR>   File "/usr/lib/python3/dist-packages/pluggy/manager.py", line 83, in <lambda>
INTERNALERROR>     self._inner_hookexec = lambda hook, methods, kwargs: hook.multicall(
INTERNALERROR>   File "/usr/lib/python3/dist-packages/pluggy/callers.py", line 208, in _multicall
INTERNALERROR>     return outcome.get_result()
INTERNALERROR>   File "/usr/lib/python3/dist-packages/pluggy/callers.py", line 80, in get_result
INTERNALERROR>     raise ex[1].with_traceback(ex[2])
INTERNALERROR>   File "/usr/lib/python3/dist-packages/pluggy/callers.py", line 187, in _multicall
INTERNALERROR>     res = hook_impl.function(*args)
INTERNALERROR>   File "/home/akim/.local/lib/python3.10/site-packages/_pytest/runner.py", line 111, in pytest_runtest_protocol
INTERNALERROR>     runtestprotocol(item, nextitem=nextitem)
INTERNALERROR>   File "/home/akim/.local/lib/python3.10/site-packages/_pytest/runner.py", line 130, in runtestprotocol
INTERNALERROR>     reports.append(call_and_report(item, "call", log))
INTERNALERROR>   File "/home/akim/.local/lib/python3.10/site-packages/_pytest/runner.py", line 221, in call_and_report
INTERNALERROR>     report: TestReport = hook.pytest_runtest_makereport(item=item, call=call)
INTERNALERROR>   File "/usr/lib/python3/dist-packages/pluggy/hooks.py", line 286, in __call__
INTERNALERROR>     return self._hookexec(self, self.get_hookimpls(), kwargs)
INTERNALERROR>   File "/usr/lib/python3/dist-packages/pluggy/manager.py", line 92, in _hookexec
INTERNALERROR>     return self._inner_hookexec(hook, methods, kwargs)
INTERNALERROR>   File "/usr/lib/python3/dist-packages/pluggy/manager.py", line 83, in <lambda>
INTERNALERROR>     self._inner_hookexec = lambda hook, methods, kwargs: hook.multicall(
INTERNALERROR>   File "/usr/lib/python3/dist-packages/pluggy/callers.py", line 203, in _multicall
INTERNALERROR>     gen.send(outcome)
INTERNALERROR>   File "/home/akim/Desktop/wazuh-qa/tests/system/conftest.py", line 132, in pytest_runtest_makereport
INTERNALERROR>     for host in host_manager.get_inventory()['managers']['hosts']:
INTERNALERROR> KeyError: 'managers'
```

</details>

Several times were executed in different machines (EC2, local) however, the error has persisted.

On the other hand, running test_cluster/test_correct_merged_file_generation/test_correct_merged_file_generation.py separated from the rest of the tests (without the pytest -m) is working ok.

Further research is required.

### PR DESCRIPTION
**Pull request by Deblintrake09**
|Related issue|
|-------------|
| closes #4763             |

## Description

This PR fixes the system tests so that it generates reports correctly when the test has the `cluster` and `one_manager_agent_env` markers.

### Updated

- tests/system/conftest.py

## Testing performed

- executed tests with marker command `# python3 -m pytest -m one_manager_agent_env -vvv --html=/home/qa/one_manager_agent.html`

```
# python3 -m pytest -m one_manager_agent_env -vvv --html=/home/qa/one_manager_agent.html
=========================================================================================== test session starts ============================================================================================
platform linux -- Python 3.10.12, pytest-7.1.2, pluggy-1.3.0 -- /usr/bin/python3
cachedir: .pytest_cache
metadata: {'Python': '3.10.12', 'Platform': 'Linux-6.6.6-76060606-generic-x86_64-with-glibc2.35', 'Packages': {'pytest': '7.1.2', 'pluggy': '1.3.0'}, 'Plugins': {'html': '3.1.1', 'testinfra': '5.0.0', 'metadata': '3.0.0'}}
rootdir: /home/qa/Descargas/wazuh-qa/tests/system, configfile: pytest.ini
plugins: html-3.1.1, testinfra-5.0.0, metadata-3.0.0
collected 192 items / 165 deselected / 27 selected                                                                                                                                                         

test_cluster/test_correct_merged_file_generation/test_correct_merged_file_generation.py::test_correct_merged_file_generation[merged_created_on_start] PASSED                                         [  3%]
test_cluster/test_correct_merged_file_generation/test_correct_merged_file_generation.py::test_correct_merged_file_generation[merged_updated_10_sec] PASSED                                           [  7%]
test_cluster/test_correct_merged_file_generation/test_correct_merged_file_generation.py::test_correct_merged_file_generation[add_non_zero_file_default_folder_check_merged_created_starts] PASSED    [ 11%]
test_cluster/test_correct_merged_file_generation/test_correct_merged_file_generation.py::test_correct_merged_file_generation[add_zero_file_default_folder_check_merged_created_starts] PASSED        [ 14%]
test_cluster/test_correct_merged_file_generation/test_correct_merged_file_generation.py::test_correct_merged_file_generation[add_non_zero_size_file_default_folder_check_merged_updated_10_sec0] PASSED [ 18%]
test_cluster/test_correct_merged_file_generation/test_correct_merged_file_generation.py::test_correct_merged_file_generation[add_non_zero_size_file_TestGroup1_folder_check_merged_updated_manager_starts] PASSED [ 22%]
test_cluster/test_correct_merged_file_generation/test_correct_merged_file_generation.py::test_correct_merged_file_generation[add_non_zero_size_file_TestGroup1_folder_check_merged_updated_10_sec] PASSED [ 25%]
test_cluster/test_correct_merged_file_generation/test_correct_merged_file_generation.py::test_correct_merged_file_generation[add_non_zero_size_file_default_folder_check_merged_updated_10_sec1] PASSED [ 29%]
test_cluster/test_correct_merged_file_generation/test_correct_merged_file_generation.py::test_correct_merged_file_generation[add_zero_size_file_TestGroup1_folder_check_merged_updated_manager_starts] PASSED [ 33%]
test_cluster/test_correct_merged_file_generation/test_correct_merged_file_generation.py::test_correct_merged_file_generation[add_zero_size_file_TestGroup1_folder_check_merged_updated_10_sec] PASSED [ 37%]
test_cluster/test_correct_merged_file_generation/test_correct_merged_file_generation.py::test_correct_merged_file_generation[add_serveral_zero_size_file_default_folder_check_merged_updated_manager_starts] PASSED [ 40%]
test_cluster/test_correct_merged_file_generation/test_correct_merged_file_generation.py::test_correct_merged_file_generation[add_serveral_zero_size_file_TestGroup1_folder_check_merged_updated_manager_start] PASSED [ 44%]
test_fim/test_files/test_files_cud.py::test_file_cud[testdir1-add] PASSED                                                                                                                            [ 48%]
test_fim/test_files/test_files_cud.py::test_file_cud[testdir1-modify] PASSED                                                                                                                         [ 51%]
test_fim/test_files/test_files_cud.py::test_file_cud[testdir1-delete] PASSED                                                                                                                         [ 55%]
test_fim/test_files/test_files_cud.py::test_file_cud[testdir2-add] PASSED                                                                                                                            [ 59%]
test_fim/test_files/test_files_cud.py::test_file_cud[testdir2-modify] PASSED                                                                                                                         [ 62%]
test_fim/test_files/test_files_cud.py::test_file_cud[testdir2-delete] PASSED                                                                                                                         [ 66%]
test_fim/test_files/test_files_cud.py::test_file_cud[testdir3-add] PASSED                                                                                                                            [ 70%]
test_fim/test_files/test_files_cud.py::test_file_cud[testdir3-modify] PASSED                                                                                                                         [ 74%]
test_fim/test_files/test_files_cud.py::test_file_cud[testdir3-delete] PASSED                                                                                                                         [ 77%]
test_fim/test_synchronization/test_synchronization.py::test_synchronization[testdir1-add-wazuh-agent1] PASSED                                                                                        [ 81%]
test_fim/test_synchronization/test_synchronization.py::test_synchronization[testdir1-add-wazuh-manager] PASSED                                                                                       [ 85%]
test_fim/test_synchronization/test_synchronization.py::test_synchronization[testdir1-modify-wazuh-agent1] PASSED                                                                                     [ 88%]
test_fim/test_synchronization/test_synchronization.py::test_synchronization[testdir1-modify-wazuh-manager] PASSED                                                                                    [ 92%]
test_fim/test_synchronization/test_synchronization.py::test_synchronization[testdir1-delete-wazuh-agent1] PASSED                                                                                     [ 96%]
test_fim/test_synchronization/test_synchronization.py::test_synchronization[testdir1-delete-wazuh-manager] PASSED                                                                                    [100%]
----------------------------------------------------------------------- generated html file: file:///home/qa/one_manager_agent.html ------------------------------------------------------------------------
======================================================================= 27 passed, 165 deselected, 5 warnings in 1750.91s (0:29:10) ========================================================================
```

| Tester             | Test path |  Local  | OS | Commit | Notes                |
|--------------------|-----------|--------|-----|--------|----------------------|
| @Deblintrake09  (Developer)  |   tests/system/        | [:green_circle:](https://github.com/wazuh/wazuh-qa/files/13717053/one_manager_agent1.zip) | Ubuntu     | a8b0ed9d281001a31ef827afbafe206a5745ccb0        | Nothing to highlight |
| @pro-akim (Reviewer)   |           | [:green_circle:](https://github.com/wazuh/wazuh-qa/files/13726489/report_one_manager_agent.zip) ⚫⚫ |        |         | Nothing to highlight |

### CODE REVIEW

File: tests/system/conftest.py [hunk 1]
```diff
@@ -126,7 +126,7 @@ def pytest_runtest_makereport(item, call):
                 arguments[key] = str(value)
         extra.append(pytest_html.extras.json(arguments, name="Test arguments"))
 
-        if "cluster" in report.markers:
+        if "cluster" in report.markers and "one_manager_agent_env" not in report.markers:
```
Comment (Rebits) [inline tests/system/conftest.py:129]: Why does `one_manager_agent_env` include the cluster marker? It seems that removing the marker would result in a more maintainable approach for this particular case.

In addition, handling the logs from cluster environment using markers is not the best approach. We should consider to change this logic in order to make it more error resistant, including a minor inventory validation

File: CHANGELOG.md [hunk 2]
```diff
@@ -17,6 +17,7 @@ All notable changes to this project will be documented in this file.
 
 ### Changed
 
+- Fix one_manager_agent_env pytest marker for System Tests ([#4763](https://github.com/wazuh/wazuh-qa/pull/4763)) \- (Tests)
```
Comment (Rebits) [inline CHANGELOG.md:20]:
```suggestion
- Fix one_manager_agent_env pytest marker for System Tests ([#4782](https://github.com/wazuh/wazuh-qa/pull/4782)) \- (Tests)
```

**Review by Rebits — CHANGES_REQUESTED** [review]
GJ, minor changes are requested

File: CHANGELOG.md (hunk 2 above)
Comment (Deblintrake09) [reply CHANGELOG.md]: Fixed in https://github.com/wazuh/wazuh-qa/pull/4782/commits/0d439954a82568453dad5ce801a6ea4a839b6653

File: tests/system/conftest.py (hunk 1 above)
Comment (Deblintrake09) [reply tests/system/conftest.py]: fixed in https://github.com/wazuh/wazuh-qa/pull/4782/commits/0baa8e303905ec25290fadc055250847f3155749

**Review by pro-akim — DISMISSED** [review]
## Review Notes
LGTM

**Pull request merged**